citest/google_session/google_session.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import requests
+
+from bs4 import BeautifulSoup
+
+class GoogleSession:
+  """Utility class for automating Google authentication.
+
+  Automates both the Google account login and scope approval to create an
+  authenticated session, and provides facilities for making authenticated
+  requests in the scope of that session.
+
+  Example usage:
+    # login_url should forward to 'accounts.google.com/ServiceLogin' page
+    login_url = "http://localhost:8084/credentials"
+    user_approval_url = "https://accounts.google.com/ServiceLoginAuth"
+    session = GoogleSession(login_url, "user@domain.net", "secret_pass_word")
+    session.get('https://google.com/cool/stuff')
+    payload = {'cool': 'stuff'}
+    session.post('https://google.com/store/me, payload)
+  """
+
+  def __init__(self,
+               login_url,
+               login,
+               pwd,
+               user_approval_url='https://accounts.google.com/ServiceLoginAuth'):
+    """Construct an instance.
+
+    Args:
+      login_url: A URL string that will eventually redirect to
+        'https://accounts.google.com/ServiceLogin'.
+      login: Username/email to login as.
+      pwd: Password for the provided login.
+      user_approval_url: The URL for a Google user to grant approval to an application.
+        Can override if necessary.
+    """
+    self.session = requests.session()
+    login_html = self.session.get(login_url)
+    soup_login = BeautifulSoup(login_html.content).find('form').find_all('input')
+    login_dict = {}
+    for u in soup_login:
```

Comment:
Why the variable name `u` here and below? It's not clear what you're iterating over.